tests/torch.test.ts

```diff
@@ -613,6 +614,180 @@ describe('Torch', function () {
+                }
+            };
+
+            /* When */
+            const method = (req: any, res: any) => {
+                /* The impl */
+            };
+            Torch(app as Application, (router: Router) => {
+                router.group({ prefix: ':foo' }, router => {
+                    router.group({ prefix: '/:bar' }, router => {
+                        router.put('/home/:id/:postId', method).where({
+                            'id': '(\\d+)',
+                            'postId': '(1|2|3)',
+                            'foo': '[0-9]+',
+                            'bar': '[a-b]'
+                        });
+                    });
+                });
+            });
+
+            /* Then */
+            expect(registeredPath).to.equal('/:foo[0-9]+/:bar[a-b]/home/:id(\\d+)/:postId(1|2|3)');
+            expect(registeredMethod).to.equal(method);
+        });
+
+        it('should not screw up the route if unmet constraints are added', function () {
+            /* Given */
+            let registeredPath: string | null = null;
+            let registeredMethod: ((req: Request, res: Response) => void) | null = null;
+
+            const app: any = {
+                delete: (path: string, middleware: Array<(req: Request, res: Response, next: NextFunction) => any>, method: ((req: Request, res: Response) => void)) => {
+                    registeredPath = path;
+                    registeredMethod = method;
+                }
+            };
+
+            /* When */
+            const method = (req: any, res: any) => {
+                /* The impl */
+            };
+            Torch(app as Application, (router: Router) => {
+                router.delete('/:id/:postId', method).where({
+                    'id': '(\\d+)',
+                    'postId': '(1|2|3)',
+                    'foo': '[0-9]+'
+                });
+            });
+
+            /* Then */
+            expect(registeredPath).to.equal('/:id(\\d+)/:postId(1|2|3)');
+            expect(registeredMethod).to.equal(method);
+        });
+
+        it('should not screw up if the order of variables is confusing', function () {
+            /* Given */
+            let registeredPath: string | null = null;
+            let registeredMethod: ((req: Request, res: Response) => void) | null = null;
+
+            const app: any = {
+                delete: (path: string, middleware: Array<(req: Request, res: Response, next: NextFunction) => any>, method: ((req: Request, res: Response) => void)) => {
+                    registeredPath = path;
+                    registeredMethod = method;
+                }
+            };
+
+            /* When */
+            const method = (req: any, res: any) => {
+                /* The impl */
+            };
+            Torch(app as Application, (router: Router) => {
+                router.delete('/:identifier/:id', method).where({
+                    'id': '(\\d+)',
+                    'identifier': '(\\d+)'
```

Comment:
Erm, this doesn't test anything, we still do not known which one is inserted where